src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -60,26 +62,35 @@ public class PullRequestHandler : IHttpHandler
             context.Response.Write(HttpUtility.HtmlEncode(result));
             return;
         }
-        else
+
+        success = false;
+
+        // Check inputs to the script. PR has precedence
+        uint pr;
+        if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
         {
-            // Check inputs to the script
-            uint pr;
-            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
-            {
-                success = SyncToPr(pr, result);
-            }
-            else
+            success = SyncToPr(pr, result);
+        }
+        else if (!string.IsNullOrEmpty(branchString))
```

Comment:
I think there's a misunderstanding on how branches work: 

* You would never attempt to checkout a PR from "a branch", because the PR implies the branch already
* When you check out a branch, you necessarily do not specify a PR number because you are asking to sync to the branch HEAD
* When you specify both, I could either output an error, or just make the best guess assumption. In this case, I assume PR take precedence
* Our CI systems will _never_ specify both

I also disagree here about `CheckPrerequisites`. All the prereqs are a necessary condition for this script to work, so I don't care about the input if (say) git is not available or the filesystem is not available. Furthermore, the later implementation of checking for branch valid input is going to rely on the script being able to run `git branch -r` and checking against a list of valid branches on `origin`. 

CleanupMergedBranches might be avoidable with invalid input, but I view CleanupMergedBranhes as an extension of CheckPrerequisites. All assumptions about checkouts being successful are essentially invalid if I cannot assume a clean branch. Hence, CleanupMergedBranches should stay where it is.